stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1501,26 +1501,27 @@ public extension Tensor {
   @inlinable
   subscript(index: Int32) -> Tensor {
     get {
-      return Raw.stridedSlice(self,
-                              begin: Tensor<Int32>([index]),
-                              end: Tensor<Int32>([index + 1]),
-                              strides: Tensor<Int32>([1])).squeezingShape(at: 0)
+      let slice = Raw.stridedSlice(
+        self, begin: Tensor<Int32>([index]), end: Tensor<Int32>([index + 1]),
+        strides: Tensor<Int32>([1]))
+      return slice.squeezingShape(at: 0)
     }
     set {
-      let left = self[0..<index]
-      let right = self[index+1..<_TFGetScalarOrDie(shapeTensor[0].handle)]
-      self = Raw.concatV2([left, newValue.rankLifted(), right], axis: Tensor<Int32>(0))
+      let count = _TFGetScalarOrDie(shapeTensor[0].handle)
```

Comment:
Would we like to change `_TFGetScalarOrDie(shapeTensor[0].handle)` to something else?
Perhaps `shape[0]` directly?

cc @rxwei @jekbradbury

Reply:
Yes please!

Reply:
Done. NFC status revoked.